homeassistant/components/vicare/number.py

```diff
@@ -0,0 +1,217 @@
+"""Number for ViCare."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from contextlib import suppress
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from PyViCare.PyViCareDevice import Device as PyViCareDevice
+from PyViCare.PyViCareDeviceConfig import PyViCareDeviceConfig
+from PyViCare.PyViCareHeatingDevice import (
+    HeatingDeviceWithComponent as PyViCareHeatingDeviceWithComponent,
+)
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ViCareRequiredKeysMixin
+from .const import DOMAIN, VICARE_API, VICARE_DEVICE_CONFIG
+from .entity import ViCareEntity
+from .utils import is_supported
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ViCareNumberEntityDescription(NumberEntityDescription, ViCareRequiredKeysMixin):
+    """Describes ViCare number entity."""
+
+    # value_getter: Callable[[PyViCareDevice], float | None] | None = None
+    value_setter: Callable[[PyViCareDevice, float], Any | None] | None = None
+    min_value_getter: Callable[[PyViCareDevice], float | None] | None = None
+    max_value_getter: Callable[[PyViCareDevice], float | None] | None = None
+    stepping_getter: Callable[[PyViCareDevice], float | None] | None = None
+
+
+PROGRAM_ENTITY_DESCRIPTIONS: tuple[ViCareNumberEntityDescription, ...] = (
+    ViCareNumberEntityDescription(
+        key="normal temperature",
+        name="Normal temperature",
```

Comment:
Use translation key. Should not be done in separate PR (that can cover the other platforms afterwards)

Reply:
The entity names are based on the number of heating circuits that are installed and get the heating circuit number appended to them. To using translation keys is not possible as far as I know.
An idea would be to have circuits in general as own devices provided by the heating device and to so also for burners and compressors.

Reply:
> The entity names are based on the number of heating circuits that are installed and get the heating circuit number appended to them

That sounds like that should be part of the device name and the entity name should only be `Normal temperature` which is appended to the device name.
I would suggest to make a preliminary PR to fix this prior to this PR